### PR TITLE
feat: implement expenses storage function

### LINKED ISSUE #65: feat: implement expenses storage function
As a careful treasurer, I can save my expenses in the app and retrieve them again the next time I use the app so that I can track expenses over a prolonged time.

### PR DESCRIPTION
**Pull request by saheer17**
This PR adds the StorageManager feature to handle saving and loading of ExpenseManager:

1. Creates storage folder/file if not present.
2. Handles errors and permission issues with user-friendly messages.
3. Adds logging for key actions and errors.
4. Adds JUnit tests for saving, loading, and missing files.
5. Adds Javadoc comments and fixes code formatting.

This feature enables reliable persistence of expenses and improves overall application stability. Resolves #65

### CODE REVIEW

File: src/main/java/seedu/orcashbuddy/Main.java [hunk 1]
```diff
@@ -76,6 +77,7 @@ private boolean executeCommand(String input) {
         try {
             Command command = parser.parse(input);
             command.execute(expenseManager, ui);
+            StorageManager.saveExpenseManager(expenseManager, ui);
```
Comment (aydrienlaw) [inline src/main/java/seedu/orcashbuddy/Main.java]: I see that you’re saving after every command execution. Would it make sense to save only after mutating commands (e.g., `add`, `delete`)? That might help reduce unnecessary disk writes and improve performance.

Maybe an approach like introducing a flag, such as `command.isMutating()`, could help decide when to save. Alternatively, save once on `bye`.

File: src/main/java/seedu/orcashbuddy/storage/StorageManager.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+     * <p>
+     * If the storage folder does not exist, it will be created. If saving fails,
+     * a user-friendly message is displayed via the {@link Ui} object.
+     *
+     * @param expenseManager the ExpenseManager object to save; must not be null
+     * @param ui             the UI object to display error messages; must not be null
+     */
+    public static void saveExpenseManager(ExpenseManager expenseManager, Ui ui) {
+        assert expenseManager != null : "ExpenseManager must not be null";
+        assert ui != null : "Ui must not be null";
```
Comment (aydrienlaw) [inline src/main/java/seedu/orcashbuddy/storage/StorageManager.java]: Nice use of assertions for defensive checks!

**Review by aydrienlaw — COMMENTED** [review]
Great update - I like how you’ve structured the persistence logic and covered it with tests!

My comments are mostly about edge cases and small refinements (saving strategy, assertions, and float comparisons). Everything else looks solid - approving this for merge!
